packages/dds/map/src/interfaces.ts

```diff
@@ -243,6 +262,17 @@ export interface IDirectoryEvents extends IEvent {
     (event: "disposed", listener: (
```

Comment:
I haven't kept up with the changes here over time - do we intend a material difference in "disposed" and "subDirectoryDeleted" (and "undisposed", "subDirectoryCreated", and "subDirectoryUndeleted") -- particularly in how a customer should respond to each?  If so, probably try to clarify the distinction our customers would need to know in the event documentation.  If not, can we eliminate down to a single one of the events for create/delete?

Reply:
The initial reason for the distinction between undispose and create was because the actions aren't necessarily the same. The create event signifies a new, empty directory, but for the rollback of a delete, this may or may not be the case - we may instead have a non-empty directory. I can add documentation for this as well, or if you think that simplifying the events to just create and delete would be better, I can do that as well. 

As for the difference between dispose/delete and undispose/undelete, that was just how the events were set up, but I plan to look into the git history to see if I can figure out the reason for the difference here. I'm also planning to dig into the Whiteboard and Loop code to see what they actually consume and if the difference matters there.

Reply:
I talked with @jatgarg and there is a distinction between "disposed" and "subDirectoryDeleted" - the disposed event marks the object unusable for the future, but the subDirectoryDeleted event actually deletes the object. I'm not entirely sure if this means that customers would have to respond to the events differently, but I can add some documentation here as well to clarify the distinction.

Reply:
Good to know that distinction between disposed and subDirDeleted. Given that, @jatgarg, does it make sense for there to be a separate event to indicate the dispose has been rolled back? or is subDirCreated/Undeleted sufficient?

Reply:
One clarification on rollback. When is a rollback allowed? If the op has already been fired for lets say delete, then can we rollback it to undelete or does the rollback only if the op is a local op and has not been fired yet? I mean how does it look to another client, does it receive two ops, one for delete and then undelete/create when another user rollback?

Reply:
In the rollback case, the original edit and the rollback remain local. Other clients will never see those ops.

Reply:
Rollback can only happen for local ops that are still in the pending state, so other clients shouldn't see the initial op (the delete) at all. I believe other clients would only see the result of the rollback.

Reply:
Users might have done something based on dispose event, so they may want to do opposite on undispose event. So, we already have undipose function, we could add the "undispose" event there. But I am in favor "subDirCreated" event rather than "Undeleted" event if that sounds good to you.